xarray/core/missing.py

```diff
@@ -706,22 +721,140 @@ def interp_func(var, x, new_x, method, kwargs):
+        # (if interpolating on a point, removes de dim)
+        final_dims = dict(
+            **{d: d for d in const_dims},
+            **{dim: _x.name for dim, _x in zip(interp_dims, new_x) if _x.size > 1},
+        )
+
+        # rename new_x to correspond to source dimension
+        def rename_index(index, dim):
+            if index.size == 1:
+                # a scalar has no name
+                return index
+            return IndexVariable(
+                dims=[dim],
+                data=index,
+                attrs=index.attrs,
+                encoding=index.encoding,
+                fastpath=True,
+            )
+
+        new_x = [rename_index(_x, dim) for dim, _x in zip(interp_dims, new_x)]
+
+        new_x_float = [_floatize_one_x(_x) for _x in new_x]
+        unsorted = any(
+            (np.any(np.diff(_x.data) < 0) for _x in new_x_float if _x.size > 1)
+        )
+        if unsorted:
+            warnings.warn(
+                "Interpolating to unsorted destination will rechunk the result",
+                da.PerformanceWarning,
+            )
+
+            sorted_idx = {
+                dim: _x.data.argsort()
+                for dim, _x in zip(interp_dims, new_x_float)
+                if _x.size > 1
+            }
+
+            new_x = [
+                _x[sorted_idx[dim]] if dim in sorted_idx else _x
+                for dim, _x in zip(interp_dims, new_x)
+            ]
+        # add missing dimensions to new_x
+        new_x = tuple([_x.set_dims(interp_dims) for _x in new_x])
+
+        # chunks x
+        x = tuple(
+            da.from_array(_x, chunks=chunks)
+            for _x, chunks in zip(x, var.chunks[nconst:])
+        )
+
+        # duplicate the ghost cells of the array in the interpolated dimensions
+        depth = {"quadratic": 2, "cubic": 3}.get(method, 1)
+        var, x, var_with_ghost, x_with_ghost = _add_interp_ghost(
+            var, x, nconst, depth=depth
+        )
+
+        # compute final chunks
+        total_chunks = _compute_chunks(x, new_x)
+        final_chunks = var.chunks[: -len(x)] + tuple(total_chunks)
+
+        # chunks new_x
+        new_x = tuple(da.from_array(_x, chunks=total_chunks) for _x in new_x)
+
+        # reshape x_with_ghost
+        # TODO: remove it (see _dask_aware_interpnd)
+        x_with_ghost = da.meshgrid(*x_with_ghost, indexing="ij")
+
+        # compute on chunks
+        res = da.map_blocks(
+            _dask_aware_interpnd,
+            var_with_ghost,
+            *x_with_ghost,
+            *new_x,
+            interp_func=func,
+            interp_kwargs=kwargs,
+            n_coords=len(x_with_ghost),
             dtype=var.dtype,
-            chunks=chunks,
-            new_axis=new_axis,
-            drop_axis=drop_axis,
+            chunks=final_chunks,
+        )
+
+        # reshape res and remove empty chunks
+        # TODO: remove it by using drop_axis and new_axis in map_blocks
+        res = res.squeeze()
```

Comment:
Is it safe to give the squeezing axis explicitly?
What happens if the original array already has a size-one dimension?

Reply:
No you're right, it's probably not safe, hence the TODO. 
But at the time I didn't manage to use `drop_axis` and `new_axis`...
I'll give an another try tomorrow.